ouroboros-network/test/Test/Ouroboros/Network/Testnet.hs

```diff
@@ -205,6 +241,404 @@ tracerDiffusionSimWithTimeName ntnAddr =
  . tracerWithTime
  $ dynamicTracer
 
+
+-- | This test coverage of ServerTrace constructors, namely accept errors.
+--
+prop_connection_manager_trace_coverage :: AbsBearerInfo
+                                       -> DiffusionScript
+                                       -> Property
+prop_connection_manager_trace_coverage defaultBearerInfo diffScript =
+
+  let sim :: forall s . IOSim s Void
+      sim = diffusionSimulation (toBearerInfo defaultBearerInfo)
+                                diffScript
+                                tracersExtraWithTimeName
+                                tracerDiffusionSimWithTimeName
+
+      events :: [ConnectionManagerTrace
+                  NtNAddr
+                  (ConnectionHandlerTrace NtNVersion NtNVersionData)]
+      events = mapMaybe (\case DiffusionConnectionManagerTrace st -> Just st
+                               _                                  -> Nothing
+                        )
+             . Trace.toList
+             . fmap (\(WithTime _ (WithName _ b)) -> b)
+             . withTimeNameTraceEvents
+                @DiffusionTestTrace
+                @NtNAddr
+             . Trace.fromList (MainReturn (Time 0) () [])
+             . fmap (\(t, tid, tl, te) -> SimEvent t tid tl te)
+             . take 125000
+             . traceEvents
+             $ runSimTrace sim
+
+      connectionManagerTraceMap
```

Comment:
Don't we have something similar in `ouroboros-network-framework` already, if so we could expose it through the `testlib`.

I realised that we might not have the `testlib` in this PR, then adding a todo / issue is enough, and we can do that after merging the other PR.

Reply:
We do have testlib in this PR! And it is a good addition!

Reply:
Done!